client/my-sites/upgrades/domain-management/dns/dns-list.jsx

```diff
@@ -43,7 +43,11 @@ const DnsList = React.createClass( {
 	addDns: function( record ) {
 		addDnsAction( this.props.selectedDomainName, record, ( error ) => {
 			if ( error ) {
-				notices.error( error.message );
+				notices.error( error.message || this.translate( 'The DNS record could not be restored.' ) );
+			} else {
+				this.props.successNotice( this.translate( 'The DNS record has been restored.' ), {
+					duration: 5000
```

Comment:
@artpi Could you clarify how you've come to that decision? I agree with Igor that a message bus was a good idea to show notices. Many of the (esp. temporary ones) notices appear out of the visual boundaries of the component; and I personally find using them within the render process as clutter. They require additional state keeping, occupies space in the render tree etc.